sphinx_gallery/docs_resolv.py

```diff
@@ -47,7 +47,7 @@ def _get_data(url):
         if encoding == 'plain':
             data = data.decode('utf-8')
         elif encoding == 'gzip':
-            data = StringIO(data)
+            data = BytesIO(data)
```

Comment:
Is the reason this doesn't work a python 2 vs. 3 thing?

Reply:
I originally thought so, but then the first commit (true TDD!) broke *every* CI, and on 3k the error is:
```
  File "/home/travis/miniconda2/envs/testenv/lib/python3.6/site-packages/sphinx_gallery-0.1.12-py3.6.egg/sphinx_gallery/docs_resolv.py", line 50, in _get_data
    data = StringIO(data)
TypeError: initial_value must be str or None, not bytes
```

Reply:
weird!

Reply:
I think the data you get over the network is supposed to be bytes in Python 3, so BytesIO makes a lot of sense to me.